relay/relay.go

```diff
@@ -451,6 +451,19 @@ func (r *Relay) handleEvents(ctx context.Context, reader2 reader.Reader, transfo
 			continue
 		}
 
+		// fake rotate event
+		if _, ok := e.Event.(*replication.RotateEvent); ok && e.Header.Timestamp == 0 && e.Header.LogPos == 0 {
+			isNew, err2 := isNewServer(ctx, r.meta.UUID(), r.db, r.cfg.Flavor)
+			if err2 != nil {
+				return err2
+			}
+			// upstream database switch
+			// report an error, let outer logic handle it
```

Comment:
where is handling logic 😵

Reply:
now it report an error，we handle it by auto resume or user resume it manually

Reply:
handle in https://github.com/pingcap/dm/blob/a778df270b62dd90adcc513f57469e8b331748ab/relay/relay.go#L201